nngt/plot/plt_networks.py

```diff
@@ -120,39 +122,84 @@ def draw_network(network, nsize="total-degree", ncolor="group", nshape="o",
-        if isinstance(restrict_sources[0], str):
+    if nonstring_container(restrict_sources): # if restrict_sources is not a string
+        if isinstance(restrict_sources[0], str): # if the first item is a string
+                                                # in this case possibly each entry
+                                                # is a group name, this must be checked
+                                                # if it is a group name, then the
+                                                # corresponding neurons' ids are added
+                                                # to the source neurons list.
             assert network.is_network(), \
                 "`restrict_sources` canbe string only for Network."
             sources = []
             for name in restrict_sources:
+                # Two cases either restrict_sources is a group name,
+                # then the corresponding neurons' ids can be automatically recovered
+                # or the restric_source is a list of neurons ids  (this is the case
+                # if we want to gather together the neurons from different groups,
+                # as for instance all the top neurons; these are all in the different groups
+                # for each top obstacle)
+                # We then have to test whether restric_sources is a valid group name 
+                assert(name in network.population.keys()),\
+                    "`restrict_sources` list contains {0} non valid group name".format(name)
+                # restrict_sources is a group name, then we recover the neurons ids
                 sources.extend(network.population[name].ids)
+
             restrict_sources = sources
-    elif isinstance(restrict_sources, str):
+    elif isinstance(restrict_sources, str): # if restrict_source is a string
+                                            # check if it is a valid group name
+        assert(restrict_sources in network.population.keys()),\
+                "`restrict_sources` list contains {0} non valid group name".format(name)
+ 
         assert network.is_network(), \
-            "`restrict_sources` canbe string only for Network."
+                "`restrict_sources` list contains {0} non valid group name".format(name)
+ 
+        # source neurons' ids recovered
         restrict_sources = network.population[restrict_sources].ids
-    if nonstring_container(restrict_targets):
-        if isinstance(restrict_targets[0], str):
+
+    if nonstring_container(restrict_targets): # if restrict_targets is not a string
+        if isinstance(restrict_targets[0], str): # check if the entries are valid group names
             assert network.is_network(), \
                 "`restrict_targets` canbe string only for Network."
+
+            # The same as above now for the targets, testing if restrict_targets
+            # is a valid group or a neuron's list 
+            if restrict_targets  in network.population.keys() :
```

Comment:
This test always returns ``False``: you are inside an `` if nonstring_container(restrict_targets)`` block, so ``restrict_targets`` is a ``list``, while all the entries in ``network.population.keys()`` are strings.

Also try to be consistent with PEP8 recommandation: do not put spaces before columns and avoid double spaces.